imap_processing/ena_maps/utils/corrections.py

```diff
@@ -0,0 +1,299 @@
+            Array of uncertainty slopes (if uncertainties provided).
+        """
+        n_levels = len(fluxes)
+        gamma = np.full(n_levels, 0, dtype=float)
+        delta_gamma = (
+            np.full(n_levels, 0, dtype=float) if uncertainties is not None else None
+        )
+
+        # Create an array of indices that can be used to create a padded array where
+        # the padding duplicates the first element on the front and the last element
+        # on the end of the array
+        extended_inds = np.pad(np.arange(n_levels), 1, mode="edge")
+
+        # Compute logs, setting non-positive fluxes to NaN
+        log_fluxes = np.log(np.where(fluxes > 0, fluxes, np.nan))
+        log_energies = np.log(energies)
+        # Create extended arrays by repeating first and last values. This allows
+        # for linear differencing to be used on the ends and central differencing
+        # to be used on the interior of the array with a single vectorized equation.
+        # Interior points use central differencing equation:
+        #     gamma_k = ln(J_{k+1}/J_{k-1}) / ln(E_{k+1}/E_{k-1})
+        # Left boundary uses linear forward differencing:
+        #     gamma_k = ln(J_{k+1}/J_{k}) / ln(E_{k+1}/E_{k})
+        # Right boundary uses linear backward differencing:
+        #     gamma_k = ln(J_{k}/J_{k-1}) / ln(E_{k}/E_{k-1})
+        log_extended_fluxes = log_fluxes[extended_inds]
+        log_extended_energies = log_energies[extended_inds]
+
+        # Extract the left and right log values to use in slope calculation
+        left_log_fluxes = log_extended_fluxes[:-2]  # indices 0 to n_levels-1
+        right_log_fluxes = log_extended_fluxes[2:]  # indices 2 to n_levels+1
+        left_log_energies = log_extended_energies[:-2]
+        right_log_energies = log_extended_energies[2:]
+
+        # Compute power-law slopes for valid indices
+        central_valid = np.isfinite(left_log_fluxes) & np.isfinite(right_log_fluxes)
+        gamma[central_valid] = (
+            (right_log_fluxes - left_log_fluxes)
+            / (right_log_energies - left_log_energies)
+        )[central_valid]
+
+        # Compute uncertainty slopes
+        if uncertainties is not None:
+            rel_unc_sq = (uncertainties / fluxes) ** 2
+            extended_rel_unc_sq = rel_unc_sq[extended_inds]
+            delta_gamma = np.sqrt(
+                extended_rel_unc_sq[:-2] + extended_rel_unc_sq[2:]
+            ) / (log_extended_energies[2:] - log_extended_energies[:-2])
+            delta_gamma[~central_valid] = 0
+
+        # Handle one-sided differencing for points where central differencing failed
+        need_fallback = ~central_valid & np.isfinite(log_fluxes)
+        # Exclude first and last points since they already use the correct
+        # one-sided differencing
+        interior_fallback = (
+            need_fallback
+            & (np.arange(n_levels) > 0)
+            & (np.arange(n_levels) < n_levels - 1)
+        )
```

Comment:
What about setting the specific indices you need? This seems a little clearer to follow for me at first glance
```suggestion
        interior_fallback = np.zeros_like(need_fallback, dtype=bool)
        interior_fallback[1:-1] = need_fallback[1:-1]
```

Reply:
Yep. I like that better. Forgot to clean up this Claude artifact.